src/options.cpp

```diff
@@ -1788,10 +1788,15 @@ void options_manager::show(bool ingame)
                 const bool is_int = cur_opt.getType() == "int";
                 const bool is_float = cur_opt.getType() == "float";
                 const std::string old_opt_val = cur_opt.getValueName();
-                const std::string opt_val = string_input_popup(
+                std::string opt_val = string_input_popup(
                                                 cur_opt.getMenuText(), 80, old_opt_val, "", "", -1, is_int);
                 if (!opt_val.empty() && opt_val != old_opt_val) {
                     if (is_float) {
+                        // Replace ',' to '.' if exist
+                        std::size_t comma_pos = opt_val.find( "," );
+                        if( comma_pos != std::string::npos ) {
+                            opt_val.replace( comma_pos, 1, "." );
+                        }
```

Comment:
I agree with @mutability: this looks wrong.

But If you still want to go this way, we have `string_replace()` for this purpose. Furthermore, a more clean solution is encapsulating whichever checks and replacements into `cOpt()`. Especially since `cOpt::getValueName()` depends on `cOpt::format` which is private. It's none of `show()`s business to know the exact type of the option in such an offhand way (`is_int` and `is_float` via comparing strings) and modify it accordingly.